packages/protocol/contracts/identity/FederatedAttestations.sol

```diff
@@ -0,0 +1,160 @@
+pragma solidity ^0.5.13;
+
+import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-solidity/contracts/utils/SafeCast.sol";
+
+import "./interfaces/IFederatedAttestations.sol";
+import "../common/interfaces/IAccounts.sol";
+import "../common/interfaces/ICeloVersionedContract.sol";
+
+import "../common/Initializable.sol";
+import "../common/UsingRegistry.sol";
+import "../common/Signatures.sol";
+import "../common/UsingPrecompiles.sol";
+import "../common/libraries/ReentrancyGuard.sol";
+
+/**
+ * @title Contract mapping identifiers to accounts
+ */
+contract FederatedAttestations is
+  IFederatedAttestations,
+  ICeloVersionedContract,
+  Ownable,
+  Initializable,
+  UsingRegistry,
+  ReentrancyGuard,
+  UsingPrecompiles
+{
+  using SafeMath for uint256;
+  using SafeCast for uint256;
+
+  struct Attestation {
```

Comment:
Nit: we might want to consider naming this struct something like "IdentifierOwnershipAttestation" so that we can more easily add other Attestation types in the future